### PR TITLE
Pipe connection fixes & Item pipe fixes

### PR DESCRIPTION
**Pull request by brachy84**
- Fixes voiding issue with round robin
- complete rewrite of enhanced round robin algorithm
- fixes the net issue that was fixed for energy net in a different pr
- fixes the distribution mode button not showing in conveyor when not something else than item pipe

- fixes rendering issue with cables by adjusting their size
- removed `AttatchmentType` and the concept of visual and actual connections
- fixes issue when pipes are not working with `doDaylightCycle false`

There is still a lot of commented code and logging, i will leave it there for debuging and remove it before merge

### CODE REVIEW

File: src/main/java/gregtech/api/pipenet/tile/TileEntityPipeBase.java [hunk 1]
```diff
@@ -296,14 +266,18 @@ public void readFromNBT(@Nonnull NBTTagCompound compound) {
...
-        NBTTagCompound blockedConnectionsTag = compound.getCompoundTag("BlockedConnectionsMap");
-        this.openConnectionsMap.clear();
-        for (String attachmentTypeKey : blockedConnectionsTag.getKeySet()) {
-            int attachmentType = Integer.parseInt(attachmentTypeKey);
-            int blockedConnections = blockedConnectionsTag.getInteger(attachmentTypeKey);
-            this.openConnectionsMap.put(attachmentType, blockedConnections);
+
+        if (compound.hasKey("Connections")) {
+            openConnections = compound.getInteger("Connections");
+        } else {
```
Comment (DStrand1) [inline src/main/java/gregtech/api/pipenet/tile/TileEntityPipeBase.java:272]: Lets do `else if (compound.hasKey("BlockedConnectionsMap")` just in case, since it could cause an NPE if it is not present

File: src/main/java/gregtech/common/covers/CoverConveyor.java [hunk 2]
```diff
@@ -473,8 +473,8 @@ public ModularUI createUI(EntityPlayer player) {
                 .setTooltipHoverString("cover.universal.manual_import_export.mode.description"));
         primaryGroup.addWidget(new ToggleButtonWidget(130, 166, 20, 20, GuiTextures.BLOCKS_INPUT, () -> blocksInput, val -> blocksInput = val).setTooltipText("cover.conveyor.blocks_input"));
 
-        TileEntity coverTile = coverHolder.getWorld().getTileEntity(coverHolder.getPos());
-        if (coverTile instanceof TileEntityItemPipe) {
+        if (coverHolder.getWorld().getTileEntity(coverHolder.getPos()) instanceof TileEntityItemPipe ||
+                coverHolder.getWorld().getTileEntity(coverHolder.getPos().offset(attachedSide)) instanceof TileEntityItemPipe) {
```
Comment (DStrand1) [inline src/main/java/gregtech/common/covers/CoverConveyor.java:477]: What's the purpose of this change? Seems like it was fine before only checking the world for a TE once

**Review by DStrand1 — COMMENTED** [review]
2 comments other than the code comments:
- Connections for Item Pipes seem very glitchy, frequently disconnecting other connections when a new one is made. Probably a simple fix, but I did not see it in a brief look. Only occurs for item pipes, not cables/wires/fluid pipes
- If the rate/s on a Conveyor is changed, and we are in Enhanced Round Robin mode, maybe we should reset the "stored state" of the round robin? Right now changing that amount can make round robin appear inconsistent, even though it is just load balancing as it should based on old data

File: src/main/java/gregtech/common/covers/CoverConveyor.java (hunk 2 above)
Comment (brachy84) [reply src/main/java/gregtech/common/covers/CoverConveyor.java:477]: Bruh. This fixes the bug where the button shows only up when the cover is on a item pipe. Now it will also show if it is on a block next to a item pipe

**Comment by Syrcan** [comment]
I found two bugs.
The first being that conveyor/pumps block the connection when connected on pipes (despite still working). Overriding `canPipePassThrough()` to `true` fixes it.

The second bug is more complicated... When a Conveyor Module is placed on an item pipe with Import mode, the first operation will yeet the cover out of existence (and client thinks it still exists).
Placing **another** Conveyor in the same spot will end up with things working as expected.

**Review by DStrand1 — APPROVED** [review]
